TheForceEngine/TFE_DarkForces/agent.cpp

```diff
@@ -434,16 +435,28 @@ namespace TFE_DarkForces
 		{
 			return JFALSE;
 		}
+		saveData->agentData.selectedMission = TFE_Endian::le32_to_cpu(saveData->agentData.selectedMission);
+		saveData->agentData.nextMission = TFE_Endian::le32_to_cpu(saveData->agentData.nextMission);
+		for (s32 i = 0; i < 140; i++)
+		{
+			saveData->ammo[i] = TFE_Endian::le32_to_cpu(saveData->ammo[i]);
+		}
 		return JTRUE;
 	}
 
-	JBool agent_writeAgentConfigData(FileStream* file, s32 agentId, const LevelSaveData* saveData)
+	JBool agent_writeAgentConfigData(FileStream* file, s32 agentId, LevelSaveData* saveData)
 	{
 		s32 fileOffset = agentId*sizeof(LevelSaveData) + s_headerSize;
 		if (!file->seek(fileOffset))
 		{
 			return JFALSE;
 		}
+		saveData->agentData.selectedMission = TFE_Endian::le32_to_cpu(saveData->agentData.selectedMission);
+		saveData->agentData.nextMission = TFE_Endian::le32_to_cpu(saveData->agentData.nextMission);
+		for (s32 i = 0; i < 140; i++)
+		{
+			saveData->ammo[i] = TFE_Endian::le32_to_cpu(saveData->ammo[i]);
+		}
```

Comment:
Since it's written to file, shouldn't these all be "cpu_to_le32()"  ?

Reply:
Sure thing, I'll add the reverse to the header for some syntactic sugar.

Reply:
I just looked closer at this function: it's used for the darkpilo.cfg file which is created by TFE, therefore you don't need any endianness conversion.

Reply:
The reason I added byteswaps here so that people can continue where they left off in the dos version. It's also easy to copy this file along with the rest of the game data which can lead to confusion if it doesn't just work.